### DISCUSSION TITLE
Suggestion: Add union type for typescript

### DISCUSSION THREAD

**Discussion by danieljarrett74** [tailwindlabs/heroicons#503] [Ideas]
I wanted to make the suggestion of adding a union type for typescript that includes all icons.  This way we can define fields for our component properties that restrict to that union type.  Would that work?

export interface IButtonProps extends React.ButtonHTMLAttributes<HTMLButtonElement> {
    icon: HeroIcon | undefined;

}

**Comment by jonasschultheiss** [comment]
yes please, I'd also appreciate this
